CRM/Contact/BAO/Query.php

```diff
@@ -5779,13 +5779,26 @@ public function getOperator() {
    * @param $having
    */
   public function filterRelatedContacts(&$from, &$where, &$having) {
-    static $_rTypeProcessed = NULL;
-    static $_rTypeFrom = NULL;
-    static $_rTypeWhere = NULL;
-
-    if (!$_rTypeProcessed) {
-      $_rTypeProcessed = TRUE;
-
+    if (!isset(Civi::$statics[__CLASS__]['related_contacts_filter'])) {
+      Civi::$statics[__CLASS__]['related_contacts_filter'] = array();
+    }
+    $_rTempCache =& Civi::$statics[__CLASS__]['related_contacts_filter'];
+    // skip if filter has already applied
+    foreach ($_rTempCache as $acache) {
+      if (strpos($from, $acache['from']) !== FALSE) {
```

Comment:
I'm just wondering about this - it seems a bit unpredictable - ie. like it would happen when it is a subset of the response & like hopefully the improved $sig_arg would make it unnecessary -  did you hit any case when it was invoked in your testing?

Reply:
Yes i received an error. for injection of select and where parts twice. This loop will prevent it.

Reply:
Unfortunately we can't do much with $sig_arg. Since there's no unique argument given.